dart/lib/flat_buffers.dart

```diff
@@ -142,13 +144,17 @@ class Builder {
   /// automatically grow the array if/as needed.  `internStrings`, if set to
   /// true, will cause [writeString] to pool strings in the buffer so that
   /// identical strings will always use the same offset in tables.
-  Builder({this.initialSize: 1024, bool internStrings = false}) {
+  Builder(
+      {this.initialSize: 1024, bool internStrings = false, Allocator allocator})
+      : _allocator = allocator ?? DefaultAllocator() {
```

Comment:
nit: I think this can just be `Allocator allocator = DefaultAllocator()` and then when we migrate to nnbd it'll be a bit clearer this can't be null.

Reply:
I've tried that originally but it didn't work, saying the default value must be "constant". See this commit for the "only" way I could get it to work: https://github.com/google/flatbuffers/pull/6397/commits/c3f1a52bdee06996df944b189f745ddecfe0691b

I'm open to suggestions